src/loaders/OmeZarrLoader.ts

```diff
@@ -254,6 +276,10 @@ class OMEZarrLoader implements IVolumeLoader {
   omeroMetadata: OmeroTransitionalMetadata;
   axesTCZYX: [number, number, number, number, number];
 
+  // TODO: should we be able to share `RequestQueue`s between loaders?
+  // TODO: store sets of requests per volume via some unique key
+  requestQueue: RequestQueue;
```

Comment:
Similar to the cache, the RequestQueue can be something of a global to the whole viewer, and created once at init time but used for any and all volumes and volumeloader instances.  I would think that the concurrencylimit should be global.
The string keys could be used with wildcards to cancel things for specific volumes (if keyed with a unique volume prefix, for example)
`cancelByPrefix(prefix: string) { for all keys if key startswith prefix then cancel } `

Reply:
An issue I'm anticipating/thinking about here is that we don't have a great way of knowing at the `SmartStoreWrapper` level which requests are associated with which load calls. All we can get out of the `Store` is chunk paths, and all we have to roughly associate them with a single `getRaw` call is the time at which they were requested. So anything we do in the future that relies on grouping chunk requests into coherent batches will run into at least some issues. Prefetching may be particularly tricky, since we have to determine based on which chunks are requested which additional chunks to load into the cache. What happens e.g. when we try to load into two volumes at the same time? How do we know when all chunk loads associated with a single `loadVolumeData` call have been queued and we can start confidently determining which chunks to prefetch?

All this makes me wonder how hard it would be to extract and modify some minimal subset of zarr.js ourselves...

Reply:
Does it seem like just making the request keys have extra information would solve this issue you're talking about?  It still sounds like something that can be dealt with by special string prefixes (or suffixes) on the request keys, and those prefixes are known by the smartstore (or the loader).  Example: instead of "path/to/chunk", use "url?path/to/chunk".  The higher level code can know how to decode that but the requestqueue doesn't care. At the very least this solves the "requested the same chunks but for separate volumes" problem.